example/lib/test_scenarios/peripheral_test_operations.dart

```diff
@@ -456,6 +492,223 @@ class PeripheralTestOperations {
+        List<Descriptor> descriptors = await chosenCharacteristic.descriptors();
+        Descriptor chosenDescriptor = descriptors.firstWhere((elem) =>
+            elem.uuid ==
+            SensorTagTemperatureUuids
+                .clientCharacteristicConfigurationDescriptor);
+
+        log("Reading value...");
+        Uint8List value = await chosenDescriptor.read();
+        log("Value $value read!");
+      });
+
+  Future<void> writeDescriptorForPeripheral({bool enable = false}) async =>
+      _runWithErrorHandling(() async {
+        log("WRITE DESCRIPTOR FOR PERIPHERAL");
+        log("Writing value...");
+        Descriptor value = await peripheral.writeDescriptor(
+          SensorTagTemperatureUuids.temperatureService,
+          SensorTagTemperatureUuids.temperatureDataCharacteristic,
+          SensorTagTemperatureUuids.clientCharacteristicConfigurationDescriptor,
+          Uint8List.fromList([enable ? 1 : 0, 0]),
+        );
+        log("Descriptor ${value.uuid} written to!");
+      });
+
+  Future<void> writeDescriptorForService({bool enable = false}) async =>
+      _runWithErrorHandling(() async {
+        log("WRITE DESCRIPTOR FOR SERVICE");
+
+        log("Fetching service");
+        List<Service> services = await peripheral.services();
+        Service chosenService = services.firstWhere((elem) =>
+            elem.uuid ==
+            SensorTagTemperatureUuids.temperatureService.toLowerCase());
+
+        log("Writing value...");
+        Descriptor value = await chosenService.writeDescriptor(
+          SensorTagTemperatureUuids.temperatureDataCharacteristic,
+          SensorTagTemperatureUuids.clientCharacteristicConfigurationDescriptor,
+          Uint8List.fromList([enable ? 1 : 0, 0]),
+        );
+        log("Descriptor ${value.uuid} written to!");
+      });
+
+  Future<void> writeDescriptorForCharacteristic({bool enable = false}) async =>
+      _runWithErrorHandling(() async {
+        log("WRITE DESCRIPTOR FOR CHARACTERISTIC");
+
+        log("Fetching service");
+        List<Service> services = await peripheral.services();
+        Service chosenService = services.firstWhere((elem) =>
+            elem.uuid ==
+            SensorTagTemperatureUuids.temperatureService.toLowerCase());
+
+        log("Fetching characteristic");
+        List<Characteristic> temperatureCharacteristics =
+            await chosenService.characteristics();
+        Characteristic chosenCharacteristic = temperatureCharacteristics.first;
+
+        log("Writing value...");
+        Descriptor value = await chosenCharacteristic.writeDescriptor(
+          SensorTagTemperatureUuids.clientCharacteristicConfigurationDescriptor,
+          Uint8List.fromList([enable ? 1 : 0, 0]),
+        );
+        log("Descriptor ${value.uuid} written to!");
+      });
+
+  Future<void> writeDescriptor({bool enable = false}) async =>
+      _runWithErrorHandling(() async {
+        log("WRITE DESCRIPTOR FOR DESCRIPTOR");
```

Comment:
So I guess the previous title wasn't a mistake. Does it mean that we read value from descriptor chosen from list of descriptors instead of calling `writeDescriptor` on characteristic? Because there is no `writeDescriptor` on descriptor. 😄

Reply:
Yeah, this is testing  `descriptor.write()`